Update template.md (#2924)
harness/developer-hub: docs/platform/13_Templates/template.md

You can make changes to the same version of the template, as long as the template's inputs remain unaltered. You must create a new version of the template for any changes in the inputs.

#### Versioning with Git Experience 

> User's can store different versions of templates on:

Reviewer: ```suggestion
You can store different versions of templates on:
```

> 1. Different branches in different repository

Reviewer: ```suggestion
1. Different branches in a different repository.
```

> 1. Different branches in a different repository.

Reviewer 2: 1 and 2 appear to mean the same to me. Just that 1 says the repo is different, but it's still the same repo as far as the different branches are concerned. Unless we are using "different" and "same" to denote a relationship with the current position of something. Or, in 1, we mean "different branches in different repositories." 

Ignore if I'm off course with my understanding of the situation and you deem this to be technically accurate.

> 1. Different branches in a different repository.

Reviewer: ```suggestion
1. Different branches in different repositories.
```

> 2. Different branches in the same repository

Reviewer: ```suggestion
2. Different branches in the same repository.
```

> 3. Same branch and same repository.

Reviewer: ```suggestion
3. The same branch and same repository.
```

> 3. The same branch and same repository.

Reviewer 2: Still wondering what "same" means here. Would "A single branch in a repository" be what was meant here?

> When selecting a template version, be sure to set the branch and repository context before switching versions. Since versions can exist in multiple branches, Harness needs to know where to render template versions.

Reviewer 2: Two references to time can be confusing: "when selecting" and "before switching versions". Would the following still be accurate?
```suggestion
Before switching to a different template version, be sure to set the branch and repository context. Since versions can exist in multiple branches, Harness needs to know where to render template versions.
```

> When selecting a version of a template, be sure to set the branch and repo context before toggling the version. The version can reside on numerous branches so Harness needs to know which location to render the version of the template from. 

Reviewer: ```suggestion
When selecting a version of a template, be sure to set the branch and repo context before toggling the version. The version can reside on multiple branches, so Harness needs to know which location to render the version of the template from. 
```

> When selecting a version of a template, be sure to set the branch and repo context before toggling the version. The version can reside on multiple branches, so Harness needs to know which location to render the version of the template from. 

Reviewer: ```suggestion
When selecting a template version, be sure to set the branch and repository context before switching versions. Since versions can exist in multiple branches, Harness needs to know where to render template versions.
```

> For more information on branching and version management with branching please see [Template Library with Git Experience docs](https://developer.harness.io/docs/platform/git-experience/configure-git-experience-for-harness-entities/#branch-selection-logic-for-fetching-referenced-entities-in-remote-pipelines)

Reviewer: ```suggestion
For more information on branching and version management with branching, go to [Template Library with Git Experience docs](https://developer.harness.io/docs/platform/git-experience/configure-git-experience-for-harness-entities/#branch-selection-logic-for-fetching-referenced-entities-in-remote-pipelines).
```
